### PR TITLE
Prepare RosConn Deliberation workshop example

### PR DESCRIPTION
**Pull request by MarcoLm993**
The main changes in this PR are:
* The introduction of a specific prefix for each ROS-related callback (listed here: https://github.com/convince-project/AS2FM/pull/32/files#diff-27c59074bf1cd9058b887ee8a2abe528c0fd679312043cfebc5144f9c6b426f1R60)
* Removal of deprecated argument for the ROS callbacks and triggers (and related adjustment of scxml files)
* Addition of action return code and support for action cancel and abort, making sure there is no clash between action goal_id and code and the various fields
* Use regex for various string-related operations, to increase robustness
* Extend array support to handle deliberation example (here: https://github.com/convince-project/AS2FM/pull/32/files#diff-27f41c294326ec06faadb5d72db2e2fec0e13edf49c94a894b767d12abb1b6d4R1 , not yet running due to unsupported BT decorators)

### CODE REVIEW

**Review by MKlauck — APPROVED** [review]
Uhh, that's a huge PR. Thank you very much for all the work 👍 Looks good to me, as far as I can tell. Not sure if we should wait for Christian's review until merging.

**Comment by MarcoLm993** [comment]
Yes, let's wait for Christian. I will also turn his example into a test, so we can consider this PR truly complete.

File: as2fm_common/src/as2fm_common/ecmascript_interpretation.py
Comment (ct2034) [inline as2fm_common/src/as2fm_common/ecmascript_interpretation.py:38]: As dicusse, I would prefer to keep the `Math.`, because thats how js works

File: scxml_converter/src/scxml_converter/scxml_entries/utils.py
Comment (ct2034) [inline scxml_converter/src/scxml_converter/scxml_entries/utils.py:69]:
```suggestion
    STATE = auto()                   # No callback (e.g. state entry/exit)
    TRANSITION = auto()              # Transition callback
    ROS_TIMER = auto()               # Timer callback
    ROS_TOPIC = auto()               # Topic callback
    ROS_SERVICE_REQUEST = auto()     # Service callback
    ROS_SERVICE_RESULT = auto()      # Service callback
    ROS_ACTION_GOAL = auto()         # Action callback
    ROS_ACTION_RESULT = auto()       # Action callback
    ROS_ACTION_FEEDBACK = auto()     # Action callback
```
This way its more maintainable. And you need `from enum import auto`.

File: scxml_converter/src/scxml_converter/scxml_entries/utils.py
Comment (ct2034) [inline scxml_converter/src/scxml_converter/scxml_entries/utils.py:19]:
```suggestion
from enum import auto, Enum
```

**Pull request merged**
